mask.json

```diff
@@ -1,14 +1,15 @@
 {
     "id": "mask",
     "summary": "Apply a mask",
-    "description": "Applies a mask to a raster data cube. A mask can either be specified as:\n\n* **Raster data cube** for which parallel pixels among `data` and `mask` are compared and those pixels in `data` are replaced, which are non-zero (for numbers) or `true` (for boolean values) in `mask`.\n* **GeoJSON or vector data cube** containing one or more polygons. All pixels for which the point at the pixel center intersects with the corresponding polygon (as defined in the Simple Features standard by the OGC) are replaced.\n\nThe pixel values are replaced with the value specified for `replacement`, which defaults to `null` (no data). No data values will be left untouched by the masking operation.",
+    "description": "Applies a mask to a raster data cube. By default, a mask can either be specified as:\n\n* **Raster data cube** for which parallel pixels among `data` and `mask` are compared and those pixels in `data` are replaced, which are non-zero (for numbers) or `true` (for boolean values) in `mask`.\n* **GeoJSON or vector data cube** containing one or more polygons. All pixels for which the point at the pixel center intersects with the corresponding polygon (as defined in the Simple Features standard by the OGC) are replaced.\n\nThis behavior can be inverted with the parameter `invert`. The pixel values are replaced with the value specified for `replacement`, which defaults to `null` (no data). No data values will be left untouched by the masking operation.",
```

Comment:
The "by default" looks a bit strange there imho.

"parallel pixels" also sounds bit strange to me, isn't "corresponding pixels" a more conventional way of explaining this?

> No data values will be left untouched by the masking operation.

Could be interpreted as "all values will be touched" :)  Also not clear if it is about `mask` or `data` values. I guess it could be rewritten as "Pixels corresponding with no-data `mask` pixels will be left untouched by the masking operation". It also only applies to the raster data cube case, so could be moved there as well?

Reply:
> intersects with the corresponding polygon

intersects with any polygon

Reply:
> The "by default" looks a bit strange there imho.

Removed.

> "parallel pixels" also sounds bit strange to me, isn't "corresponding pixels" a more conventional way of explaining this?

Not sure, I think I took this wording from another library originally.

> Also not clear if it is about `mask` or `data` values.

Hmm, I think the reason was, because it applies to both, but I'm not sure about this any longer. What makes sense from the users perspective?

> It also only applies to the raster data cube case, so could be moved there as well?

Only if it only applies to the mask, but I think it was meant to also apply on the data itself (see above).

Reply:
> What makes sense from the users perspective?

I would expect that only `mask` defines the masking:

- mask == true : replace
- mask == false : preserve `data
- mask == null : preserve `data`

and in invert mode:

- mask == true: preserve
- mask == false: replace
- mask == null : preserve `data`

Reply:
What happens with no data values in data? I think I would expect no data to stay as no data, but not sure this is what usually is expected. Or should we add "ignore_nodata" as in other processes?

Reply:
I would expect that no-data is replaced with the replacement value if the mask dictates so. The null-ness of the `data` value is irrelevant I would think.

Reply:
Okay, thanks. More opinions on this?

Reply:
Now that there's a proposal for the parameter "expression" in mask, should we leave it up to the user to decide whether to care for null in the mask or not?

Reply:
> should we leave it up to the user to decide whether to care for null in the mask or not?

the output of the expression could still contain null I guess, so the backend still has to handle that case in a consistent way, no?

Reply:
Indeed, I forgot about that. I'll add documentation for that case.

Reply:
Although, it is implicitly clear:

> A pixel is replaced in `data` if the expression returns `true`

That means for false and null nothing changes...